File: medium/Q1/replace.py
import collections


def replace_string(string):
    """ Function to replace characters that repeat more than once """
    string = string.lower()
    new_dict = dict(
        collections.Counter(string)
    )  # Counter method counts the number of occurence of each character
    for char in string:
        if new_dict[char] > 1:
            string = string.replace(char, "*")
    return string


if __name__ == "__main__":
    # Test cases
    print(replace_string("Missisippi") == ("m*********"))  # True
    print(replace_string("Computer") == ("C*mp*t*r"))  # False
    print(replace_string("MisSisIppi") == ("m*********"))  # True
    print(
        replace_string("Hello, World! I love C++") == ("h*ll*,*w*rld!*i***v**c**")
    )  # False

File: medium/Q1/replace.js
// Function to replace characters that repeat more than once using regular expression
const replaceString = (string) => {
  new_arr = [];
  string = string.toLowerCase(); // Converting string to lowercase
  for (char of string) {
    var regexp = new RegExp(`[^${char}]`, "g"); // Storing regexp values as RegExp Object
    if (string.replace(regexp, "").length > 1) {
      /* Pushing regexp that occur more than once */
      new_arr.push(String(regexp).replace(regexp, ""));
    }
    if (new Set(new_arr).has(char)) {
      // Coverting array to set and then replacing all occurence
      string = string.replace(char, "*");
    }
  }
  return string;
};

console.log(replaceString("Missisippi") === "m*********"); // true
console.log(replaceString("Computer") === "C*mp*t*r"); // false
console.log(replaceString("MisSisIpPi") === "m*********"); // true
console.log(
  replaceString("Hello, World! I love C++") === "h*ll*,*w*rld!*i***v**c**"
); // false
